cpp/ql/test/experimental/query-tests/Security/CWE/CWE-788/semmle/tests/test.c

```diff
@@ -0,0 +1,44 @@
+struct buffers
+{
+    unsigned char buff1[50];
+    unsigned char *buff2;
+} globalBuff1,*globalBuff2,globalBuff1_c,*globalBuff2_c;
+
+
+void badFunc0(){
+  unsigned char buff1[12];
+  struct buffers buffAll;
+  struct buffers * buffAll1;
+  
+  buff1[strlen(buff1)]=0;
```

Comment:
Could you please annotate these test cases with the following comments:
- `GOOD` when there is no problem with the assignment, and the query _does not_ flag it.
- `BAD` when there _is_ a problem with the assignment, and the query _does_ flag it.
- `BAD [NOT DETECTED]` when there _is_ a problem with the assignment, but the query _does not_ flag it.
- `GOOD [FALSE POSITIVE]` when there _is not_ a problem with the assignment, but the query _does_ flag it.

You can see an example of these comments here: https://github.com/github/codeql/blob/main/cpp/ql/test/experimental/query-tests/Security/CWE/CWE-190/AllocMultiplicationOverflow/test.cpp